src/jit/gentree.h

```diff
@@ -2357,9 +2432,10 @@ struct GenTreeCall final : public GenTree
     CORINFO_SIG_INFO* callSig;                // Used by tail calls and to register callsites with the EE
 
     regMaskTP         gtCallRegUsedMask;      // mask of registers used to pass parameters
+
 #ifdef FEATURE_UNIX_AMD64_STRUCT_PASSING
-    SYSTEMV_AMD64_CORINFO_STRUCT_REG_PASSING_DESCRIPTOR structDesc;
-#endif // FEATURE_UNIX_AMD64_STRUCT_PASSING
+    MultiRegReturnTypeDesc    gtMultiRegReturnTypeDesc;
+#endif 
```

Comment:
The point is not to **_replace_** gtType, but rather to **_add_** the information about how the type is actually passed, and to stop lying about the real type of the node. For calls, the gtType would be the **_actual_** return type (e.g. TYP_STRUCT), but the gtReturnDesc would describe the mechanics for how it is returned.